### PR TITLE
Add legal copy and confirmation checkbox

### PR DESCRIPTION
**Pull request by kazbaig**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/web-ui/src/partials/TextAlerts/TextAlerts.vue [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+
+        <button type="submit" class="submit btn" :disabled="!hasConsented">Submit</button>
+      </div>
+
+      <div class="consent d-flex align-items-start text-left">
+        <input type="checkbox" class="consent-checkbox mr-2" id="text-alerts-consent" v-model="hasConsented" />
+        <label class="" for="text-alerts-consent"
+          >I consent to receive automated text messages (including marketing messages) from or on behalf of Amazon Web
+          Services about the Retail Demo Store, including Amazon Pinpoint, at my mobile number above. Consent is not a
+          condition of any purchase. Message and data rates may apply.</label
```
Comment (alexchirayath) [inline src/web-ui/src/partials/TextAlerts/TextAlerts.vue:26]: Please add in the text around on how the data entered will only be used for demonstrating features of messaging/email within the demo experience

File: src/web-ui/src/partials/TextAlerts/TextAlerts.vue [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
-      <button type="submit" class="submit btn">Submit</button>
+    <p class="mb-3">
+      Enter your mobile number to receive texts about the Retail Demo Store, including Amazon Pinpoint.
+    </p>
+    <form @submit.prevent="onSubmit" class="mb-2 form">
+      <div class="mb-2 d-flex justify-content-center align-items-stretch">
+        <input
+          type="tel"
+          name="phoneNumber"
+          placeholder="Enter cellphone number"
```
Comment (alexchirayath) [inline src/web-ui/src/partials/TextAlerts/TextAlerts.vue:12]: I am wondering if we could replace the placeholder to the US cellphone number format?
Ex: +1-123-456-7890

File: src/web-ui/src/partials/TextAlerts/TextAlerts.vue (hunk 1 above)
Comment (kazbaig) [reply src/web-ui/src/partials/TextAlerts/TextAlerts.vue]: Done

File: src/web-ui/src/partials/TextAlerts/TextAlerts.vue (hunk 2 above)
Comment (alexchirayath) [reply src/web-ui/src/partials/TextAlerts/TextAlerts.vue:12]: Resolving comment as per internal discussion

**Review by alexchirayath — APPROVED** [review]
LGTM

**Pull request merged**